mavros_extras/src/plugins/debug_value.cpp

```diff
@@ -0,0 +1,268 @@
+public:
+	DebugValuePlugin() : PluginBase(),
+		debug_nh("~debug"),
+		publish_debug(false)
+	{ }
+
+	void initialize(UAS &uas_)
+	{
+		PluginBase::initialize(uas_);
+
+		// subscribers
+		debug_sub = debug_nh.subscribe("send", 10, &DebugValuePlugin::debug_cb, this);
+
+		debug_nh.param("publish_debug", publish_debug, false);
+
+		// publishers (optional)
+		if (publish_debug) {
+			debug_pub = debug_nh.advertise<mavros_msgs::DebugValue>("debug", 10);
+			/** @todo create a pub config based on the naming of the debug msgs,
+			 * using a map_dict as in distance_sensor plugin
+			 */
+			debug_vector_pub = debug_nh.advertise<mavros_msgs::DebugValue>("debug_vector", 10);
+			named_value_float_pub = debug_nh.advertise<mavros_msgs::DebugValue>("named_value_float", 10);
+			named_value_int_pub = debug_nh.advertise<mavros_msgs::DebugValue>("named_value_int", 10);
+		}
+	}
+
+	Subscriptions get_subscriptions() {
+		return {
+			       make_handler(&DebugValuePlugin::handle_debug),
+			       make_handler(&DebugValuePlugin::handle_debug_vector),
+			       make_handler(&DebugValuePlugin::handle_named_value_float),
+			       make_handler(&DebugValuePlugin::handle_named_value_int)
+		};
+	}
+
+private:
+	ros::NodeHandle debug_nh;
+
+	ros::Subscriber debug_sub;
+
+	ros::Publisher debug_pub;
+	ros::Publisher debug_vector_pub;
+	ros::Publisher named_value_float_pub;
+	ros::Publisher named_value_int_pub;
+
+	// Decide if we publish the debug data to topics or not
+	bool publish_debug;
+
+	/* -*- message handlers -*- */
+
+	/**
+	 * @brief Handle DEBUG message.
+	 * Message specification: http://mavlink.org/messages/common/#DEBUG
+	 * @param msg	Received Mavlink msg
+	 * @param debug	DEBUG msg
+	 */
+	void handle_debug(const mavlink::mavlink_message_t *msg, mavlink::common::msg::DEBUG &debug)
+	{
+		auto debug_msg = boost::make_shared<mavros_msgs::DebugValue>();
+
+		ros::Time stamp = m_uas->synchronise_stamp(debug.time_boot_ms);
+
+		ROS_DEBUG_STREAM_COND_NAMED(!publish_debug, "DEBUG", "timestamp: " << stamp
+										   << " index: " << debug.ind
+										   << " value: " << debug.value);
+
+		if (publish_debug) {
+			debug_msg->header.stamp = stamp;
+			debug_msg->type = debug_msg->TYPE_DEBUG;
+			debug_msg->index = debug.ind;
+			debug_msg->value_float = debug.value;
+
+			debug_pub.publish(debug_msg);
+		}
+	}
+
+	/**
+	 * @brief Handle DEBUG_VECT message.
+	 * Message specification: http://mavlink.org/messages/common/#DEBUG_VECT
+	 * @param msg	Received Mavlink msg
+	 * @param debug	DEBUG_VECT msg
+	 */
+	void handle_debug_vector(const mavlink::mavlink_message_t *msg, mavlink::common::msg::DEBUG_VECT &debug)
+	{
+		auto debug_msg = boost::make_shared<mavros_msgs::DebugValue>();
+
+		std::string name = mavlink::to_string(debug.name);
+		ros::Time stamp = m_uas->synchronise_stamp(debug.time_usec);
+
+		std::vector<float> vec = {debug.x, debug.y, debug.z};
+
+		ROS_DEBUG_STREAM_COND_NAMED(!publish_debug, "DEBUG_VECT", "timestamp: " << stamp
```

Comment:
No! You can't deduce which plugin and what name used in resulting log msg.
That is why i added prefixes. Use common rule logging name = plugin name.
I'm pretty sure that per-message level control is not needed.